ipc/wallet/src/evm/btc_keys.rs

```diff
@@ -0,0 +1,49 @@
+use anyhow::anyhow;
+use anyhow::Result;
+
+use crate::evm::KeyInfo;
+use libsecp256k1::{PublicKey, SecretKey};
+
+pub const DEFAULT_BTC_KEYSTORE_NAME: &str = "btc_keystore.json";
+
+/// Generate a random secp256k1 private key, subject to the constraint that the
+/// public key partity is even.
+pub fn random_btc_secret_key() -> KeyInfo {
+    let secret_key = loop {
```

Comment:
I think we can negate the secret key to negate its parity, instead of looping. But the `libsecp256k1` doesn't provide the negate method (the underlying Scalar does but it's not public, annoying). So let's keep it this way.